packages/flutter/test/cupertino/scaffold_test.dart

```diff
@@ -25,6 +25,31 @@ void main() {
     expect(tester.getTopLeft(find.byType(Center)), const Offset(0.0, 0.0));
   });
 
+testWidgets('Opaque bar pushes contents down', (WidgetTester tester) async {
+    BuildContext childContext;
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: MediaQuery(
+        data: const MediaQueryData(viewInsets: EdgeInsets.only(top: 20)),
+        child: CupertinoPageScaffold(
+          navigationBar: const CupertinoNavigationBar(
+            middle: Text('Opaque'),
+            backgroundColor: Color(0xFFF8F8F8),
+          ),
+          child: Builder(
+            builder: (BuildContext context) {
+              childContext = context;
+              return Container();
+            },
+          ),
+        ),
+      ),
+    ));
+
+    expect(MediaQuery.of(childContext).padding.top, 0);
+    expect(tester.getRect(find.byType(Container)), Rect.fromLTRB(0, 44, 800, 600));
```

Comment:
Add comment here: e.g. the top is at 44px from the top because it's pushed down by the nav bar but the nav bar absorbed the 20px media query padding.